CODE_REVIEW.md

```diff
@@ -0,0 +1,38 @@
+**Code Smells**
+
+1. The naming convention of variables for below files are not proper.
+    `book-search.component.html`
+    `book-search.component.ts`
+    `reading-list.component.html`
+2.  In `book-search.component.html`, `submit` event was added for submit form but as it is reactive form the best practice is to use ngSubmit which provides form validation before sending the request.
+3.  In file `book-search.component.html`, the template contains a date function to change the format of the date. 
+4. To prevent memory leak we can use an async pipe instead of subscription, for retrieving books. By adding '| async' pipes the subscriptions are handled automatically so there is no need to unsubscribe manually.
+5. In file `total-count.component.ts`, we can remove the ngOnInit() lifecycle since it is not utilized. We should not import Angular lifecycle hooks without their utilization.
+6. In file `book-search.component.html`, for proper accessibility, buttons should be represented with a <button> tag rather than an anchor tag with a (click) listener. An anchor tag is primarily used to create links between pages. While a button is used for on page functionality.
+
+**Improvements**
+
+1. Fixed the naming conventions for variables in above mentioned files.
+2. Changed `submit` to `ngSubmit` as it is best practice.
+3. Replaced the date function with Angular Date pipe.
```

Comment:
Give reason why it's recommended to use Date pipe vs method

Reply:
Reason provided.